src/storage/src/memtable/btree.rs

```diff
@@ -51,7 +52,39 @@ impl BTreeMemtable {
             id,
             schema,
             map: Arc::new(RwLock::new(BTreeMap::new())),
-            estimated_bytes: AtomicUsize::new(0),
+            stats: Default::default(),
+        }
+    }
+
+    /// Updates memtable stats.
+    /// This function is guarded by `BTreeMemtable::map` so that store-after-load is safe.
+    fn update_stats(&self, min: Option<Value>, max: Option<Value>) {
+        if let Some(min) = min {
+            let min_val = match min {
+                Value::Int64(v) => v,
+                Value::Timestamp(t) => t.value(),
+                _ => unreachable!(),
+            };
```

Comment:
nitpick: Maybe we should a method (`as_timestamp_value()` or something else) to `Value` or just define a helper function to do it in this mod.